tests/arch/arm64/arm_psci/src/main.c

```diff
@@ -24,12 +25,8 @@ void test_psci_func(void)
 		       PSCI_VERSION_MINOR(ver) < 2),
 		       "Wrong PSCI firware version");
 
-	/* This should return 0: (one core in the affinity instance is ON) */
-	ret = psci_affinity_info(psci, 0, 0);
-	zassert_true(ret == 0, "Wrong return code from psci_affinity_info");
-
 	/* This should return -PSCI_RET_ALREADY_ON that is mapped to -EINVAL */
-	ret = psci_cpu_on(psci, 0, 0);
+	ret = pm_cpu_on(psci, 0, 0);
```

Comment:
To operate on cpu what the user has to do ?
Get the device handle to a "pm compatible device"  which can be PSCI or something else.
and operate on it.
Instead should it just operate on the apis like `pm_cpu_ops.cpu_on(cpuid)`

The change required will be to maintain `pm_cpu_ops`  struct and provide registration api. 
The ops provider may or may not be a device. choosing the ops provider is a platform config choice. CONFIG_PSCI etc. The is conventional approach.

I want to see pros in this new approach as well if we have to deviate from the conventional approach. Thoughts?

Reply:
> To operate on cpu what the user has to do ?
> Get the device handle to a "pm compatible device" which can be PSCI or something else.
> and operate on it.

Correct.

> Instead should it just operate on the apis like `pm_cpu_ops.cpu_on(cpuid)`
> 
> The change required will be to maintain `pm_cpu_ops` struct and provide registration api.
> The ops provider may or may not be a device. choosing the ops provider is a platform config choice. CONFIG_PSCI etc. The is conventional approach.
> 
> I want to see pros in this new approach as well if we have to deviate from the conventional approach. Thoughts?

Another hybrid approach can be something like you proposed but leveraging the DTS `enable-method` of the cpu nodes property to register the proper driver / mechanism.